src/test/java/apitests/DriverTest.java

```diff
@@ -0,0 +1,187 @@
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = Application.class)
+@WebIntegrationTest
+public class DriverTest {
+
+	private MockMvc mockMvc;
+
+	@Autowired
+	private DriverDAO driverManager;
+
+	@Autowired
+	private AddressDAO addressManager;
+
+	private ArrayList<Driver> driverList = new ArrayList<Driver>();
+	private ArrayList<Address> addressList = new ArrayList<Address>();
+
+	@Autowired
+	private WebApplicationContext webApplicationContext;
+
+	Driver driver1;
+	Driver driver2;
+	
+	Driver testDriver1;
+	Driver testDriver2;
+	
+	private MediaType contentType = new MediaType(MediaType.APPLICATION_JSON.getType(),
+			MediaType.APPLICATION_JSON.getSubtype(), Charset.forName("utf8"));
+
+	@Before
+	public void setup() throws Exception {
+
+		mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build();
+		this.addressList.add(addressManager.add(TestModelsFactory.createTestAddress1()));
+		driver1 = driverManager.add(TestModelsFactory.createTestDriver1(addressList.get(0)));
+		driverList.add(driver1);
+		driver2 = driverManager.add(TestModelsFactory.createTestDriver2(addressList.get(0)));
+		driverList.add(driver2);
+
+	}
+
+	@After
+	public void cleanup() throws Exception {
+
+		for (Driver driver : driverList) {
+			driverManager.delete(driver);
+		}
+	}
+
+	@Test
+	public void getDriver() throws Exception {
+		
+		testDriver1 = driverList.get(1);
+				
+		mockMvc.perform(get("/api/drivers/" + testDriver1.getId()))
+				.andExpect(status().isOk())
+				.andExpect(content().contentType(contentType))
+				.andExpect(jsonPath("$.id", is( (int) testDriver1.getId() )))
+				.andExpect(jsonPath("$.firstName", is(testDriver1.getFirstName())))
+				.andExpect(jsonPath("$.lastName", is(testDriver1.getLastName())))
+				.andExpect(jsonPath("$.salary", is(testDriver1.getSalary().doubleValue())))
+				.andExpect(jsonPath("$.pesel", is(testDriver1.getPESEL())))
+				.andExpect(jsonPath("$.available", is(testDriver1.isAvailable())))
+				.andExpect(jsonPath("$.deleted", is(testDriver1.isDeleted())));
+	}
+
+	@Test
+	public void postDriver() throws Exception {
+		
+		Driver json = TestModelsFactory.createTestDriver3(addressList.get(0));
+		
+		mockMvc.perform(post("/api/drivers/")
+				.contentType(contentType).content(convertObjectToJsonBytes(json)))
+				.andExpect(status().isOk())
+				.andExpect(content().contentType(contentType))
+				.andExpect(jsonPath("$.id", is((int) driver2.getId() + 1)))
+				.andExpect(jsonPath("$.firstName", is("Jan")))
+				.andExpect(jsonPath("$.lastName", is("Czapla")))
+				.andExpect(jsonPath("$.salary", is(3200.00)))
+				.andExpect(jsonPath("$.pesel", is("12345687911")))
+				.andExpect(jsonPath("$.available", is(true)))
+				.andExpect(jsonPath("$.deleted", is(false)));
```

Comment:
Instead of checking hardocoded values use Driver object getters e.g. (Assuming that we are adding driver object accessible under `json` variable):
```
.andExpect(jsonPath("$.firstName", is(json.getFirstName())))
```
etc.